nibabel/streamlines/tractogram.py

```diff
@@ -432,11 +432,8 @@ def apply_affine(self, affine, lazy=False):
         if np.all(affine == np.eye(4)):
             return self  # No transformation.
 
-        BUFFER_SIZE = 10000000  # About 128 Mb since pts shape is 3.
-        for start in range(0, len(self.streamlines.data), BUFFER_SIZE):
-            end = start + BUFFER_SIZE
-            pts = self.streamlines._data[start:end]
-            self.streamlines.data[start:end] = apply_affine(affine, pts)
+        for i in range(len(self.streamlines)):
+            self.streamlines[i] = apply_affine(affine, self.streamlines[i])
```

Comment:
Does this not cause a significant slowdown, to only update one streamline at a time?

Reply:
It didn't show up when running `benchmark_streamlines.py` but to be fair that benchmark might not be realistic. I don't have any large tractogram file I could use to benchmark it. Do you know where I could easily get one? (or @skoudoro maybe knows ?).

Reply:
I supposed you already downloaded one. You can look in your home folder:  `.dipy/bundle_atlas_hcp842/Atlas_80_Bundles/whole_brain/whole_brain_MNI.trk`

Reply:
And you checked that this was fine?

Reply:
It's a bit slower but at least it doesn't have side-effects.
Benchmark: loading the `whole_brain_MNI.trk` five times.
```
Old: Loaded 144,678 streamlines 5 times in  16.57
New: Loaded 144,678 streamlines 5 times in  19.61
```